server/src/main/kotlin/suwayomi/tachidesk/global/model/table/GlobalMetaTable.kt

```diff
@@ -8,11 +8,27 @@ package suwayomi.tachidesk.global.model.table
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 import org.jetbrains.exposed.dao.id.IntIdTable
+import suwayomi.tachidesk.global.impl.GlobalUpdateSettingsDataClass
 
 /**
  * Metadata storage for clients, server/global level.
  */
 object GlobalMetaTable : IntIdTable() {
     val key = varchar("key", 256)
+    val type = varchar("type", 256)
     val value = varchar("value", 4096)
 }
+
+inline fun <reified DataClass> GlobalMetaTable.toDataClass(settingsMap: Map<String, String>) = run {
+    val getSettingValue = { setting: String -> settingsMap[setting] ?: throw IllegalArgumentException() }
+
+    when (DataClass::class) {
+        GlobalUpdateSettingsDataClass::class -> GlobalUpdateSettingsDataClass(
+            getSettingValue("excludeUnreadChapters").toBoolean(),
+            getSettingValue("excludeNotStarted").toBoolean(),
+            getSettingValue("excludeCompleted").toBoolean(),
+        ) as DataClass
+
+        else -> throw IllegalStateException("Unknown DataClass Type")
+    }
+}
```

Comment:
these don't belong in this table as discussed earlier.
You can make a MangaMeta prefix for classes.

Reply:
overall in the GlobalMeta table (the database table) or not adding the toDataClass funciton to the table class?

Reply:
what do you have in mind?